### PR TITLE
E2E tests: Fix Brute Force tests

### PR DESCRIPTION
**Pull request by mauromalara**
|Related issue|
|-------------|
| #3208 |

## Description

I fixed the `test_brute_force_rdp` test by specifying the version of Hydra to be installed (v9.2).

Also, I added a role task to search for the alert before fetching the `alerts.json`.

### Added

- `search_alert.yaml` role task

### Updated

- "Generate events" playbooks
- Test cases files

---

## Testing performed

| Results | Note
| -- | --
| [🟢](https://github.com/wazuh/wazuh-qa/files/9470844/3208-mauromalara.zip) (6 rounds) | In R2 and R6 RDP fails, so I fixed the error by removing the `failed_when` condition because it's unnecessary (the command is executed but ansible does not wait for it to finish)
| [🟢](https://github.com/wazuh/wazuh-qa/files/9471195/3208-R6-mauromalara.zip) | 6 rounds of RDP only to check the change mentioned in the previous note.

### CODE REVIEW

File: tests/end_to_end/test_basic_cases/test_brute_force/test_brute_force_ssh/data/playbooks/generate_events.yaml [hunk 1]
```diff
@@ -33,14 +33,23 @@
...
         - "'Permission denied' not in result.stdout"
 
-    - name: Wait for alert
-      wait_for:
-        timeout: 5
-
 - name: Get alerts file
   hosts: managers
+  vars:
+    ts:
```
Comment (jmv74211) [inline tests/end_to_end/test_basic_cases/test_brute_force/test_brute_force_ssh/data/playbooks/generate_events.yaml:39]:
```suggestion
    timestamp:
```

File: tests/end_to_end/test_basic_cases/test_brute_force/test_brute_force_ssh/data/playbooks/generate_events.yaml [hunk 2]
```diff
@@ -33,14 +33,23 @@
...
   tasks:
 
+    - name: Search alert in alerts log
+      include_role:
+        name: manage_alerts
+        tasks_from: search_alert.yaml
+      vars:
+        ts: \d+-\d+-\d+T\d+:\d+:\d+\.\d+[+|-]\d+
+        custom_regex: "{\"timestamp\":\"{{ ts }}\",\"rule\":{\"level\":{{ rule_level }},\
```
Comment (jmv74211) [inline tests/end_to_end/test_basic_cases/test_brute_force/test_brute_force_ssh/data/playbooks/generate_events.yaml:48]:
```suggestion
        custom_regex: "{\"timestamp\":\"{{ timestamp }}\",\"rule\":{\"level\":{{ rule_level }},\
```

File: tests/end_to_end/test_basic_cases/test_brute_force/test_brute_force_rdp/data/playbooks/generate_events.yaml [hunk 3]
```diff
@@ -22,18 +22,22 @@
...
 - name: Get alerts file
   hosts: managers
   tasks:
 
+    - name: Search alert in alerts log
+      include_role:
+        name: manage_alerts
+        tasks_from: search_alert.yaml
+      vars:
+        ts: \d+-\d+-\d+T\d+:\d+:\d+\.\d+[+|-]\d+
```
Comment (jmv74211) [inline tests/end_to_end/test_basic_cases/test_brute_force/test_brute_force_rdp/data/playbooks/generate_events.yaml:35]:
```suggestion
        timestamp: \d+-\d+-\d+T\d+:\d+:\d+\.\d+[+|-]\d+
```

File: tests/end_to_end/test_basic_cases/test_brute_force/test_brute_force_rdp/data/playbooks/generate_events.yaml [hunk 4]
```diff
@@ -22,18 +22,22 @@
...
+        custom_regex: "{\"timestamp\":\"{{ ts }}\",\"rule\":{\"level\":{{ rule_level }},\
```
Comment (jmv74211) [inline tests/end_to_end/test_basic_cases/test_brute_force/test_brute_force_rdp/data/playbooks/generate_events.yaml:36]:
```suggestion
        custom_regex: "{\"timestamp\":\"{{ timestamp }}\",\"rule\":{\"level\":{{ rule_level }},\
```

**Review by jmv74211 — CHANGES_REQUESTED** [review]

File: tests/end_to_end/test_basic_cases/test_brute_force/test_brute_force_ssh/data/playbooks/generate_events.yaml (hunk 1 above)
Comment (jmv74211) [reply tests/end_to_end/test_basic_cases/test_brute_force/test_brute_force_ssh/data/playbooks/generate_events.yaml:39]: Is necessary to define this null var at this level?

File: tests/end_to_end/test_basic_cases/test_brute_force/test_brute_force_rdp/data/playbooks/generate_events.yaml (hunk 4 above)
Comment (mauromalara) [reply tests/end_to_end/test_basic_cases/test_brute_force/test_brute_force_rdp/data/playbooks/generate_events.yaml]: Done in 4cebf5d

File: tests/end_to_end/test_basic_cases/test_brute_force/test_brute_force_rdp/data/playbooks/generate_events.yaml (hunk 3 above)
Comment (mauromalara) [reply tests/end_to_end/test_basic_cases/test_brute_force/test_brute_force_rdp/data/playbooks/generate_events.yaml]: Done in 4cebf5d

File: tests/end_to_end/test_basic_cases/test_brute_force/test_brute_force_ssh/data/playbooks/generate_events.yaml (hunk 2 above)
Comment (mauromalara) [reply tests/end_to_end/test_basic_cases/test_brute_force/test_brute_force_ssh/data/playbooks/generate_events.yaml]: Done in 4cebf5d

File: tests/end_to_end/test_basic_cases/test_brute_force/test_brute_force_ssh/data/playbooks/generate_events.yaml (hunk 1 above)
Comment (mauromalara) [reply tests/end_to_end/test_basic_cases/test_brute_force/test_brute_force_ssh/data/playbooks/generate_events.yaml]: No, deleted in 4cebf5d

**Review by jmv74211 — APPROVED** [review]

**Pull request merged**